cmd/kots/cli/install.go

```diff
@@ -444,11 +448,22 @@ func InstallCmd() *cobra.Command {
 				switch status {
 				case storetypes.VersionPendingPreflight:
 					log.ActionWithSpinner("Waiting for preflight checks to complete")
+					log.FinishSpinner()
 					if err := ValidatePreflightStatus(deployOptions, authSlug, apiEndpoint); err != nil {
-						log.FinishSpinnerWithError()
-						return errors.Wrap(err, "failed to validate preflight results")
+						log.Errorf("Preflight checks contain warnings or errors. The application was not deployed")
+						perr := preflightError{}
+						if errors.As(err, &perr) {
+							cmd.SilenceErrors = true
+							var s strings.Builder
+							s.WriteString("\nPreflight check results (state - title - message)")
```

Comment:
We have a `pkg/print` where have logic for pretty-printing various things.  I think [`pkg/print/config.go`](https://github.com/replicatedhq/kots/blob/main/pkg/print/config.go) could be a good reference since those print config validation errors, which might be similar to something like preflight warnings/errors. This logic could live there.